### PR TITLE
Issue265: Progress Monitors

### PR DESCRIPTION
**Pull request by JRustyHaner**
#265

### CODE REVIEW

File: mofacts/client/views/experimentReporting/studentReporting.js [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
 import {INVALID} from '../../../common/Definitions';
 import {meteorCallAsync} from '../..';
+import gauge, {
+  Gauge,
+  Donut,
+  BaseDonut,
+  TextRenderer
+
+} from '../../lib/gauge.js';
+import { Session } from 'meteor/session';
```
Comment (andrewtackett) [inline mofacts/client/views/experimentReporting/studentReporting.js:12]: Is this necessary? I believe this is a meteor framework global, but I may have imported it to get rid of linting errors in other files

File: mofacts/client/views/experimentReporting/studentReporting.js [hunk 2]
```diff
@@ -36,37 +81,40 @@ Template.studentReporting.rendered = async function() {
   // let dataAlreadyInCache = false;
   if (Roles.userIsInRole(Meteor.user(), ['admin', 'teacher'])) {
     console.log('admin/teacher');
-  } else {
-    Session.set('curStudentID', studentID);
-    Session.set('studentUsername', studentUsername);
-  }
-  if (Session.get('instructorSelectedTdf')) {
-    Tracker.afterFlush(async function() {
+  } else {scorer.afterFlush(async function() {
```
Comment (andrewtackett) [inline mofacts/client/views/experimentReporting/studentReporting.js:84]: Was this autocorrected from something like Timer?

File: mofacts/client/views/experimentReporting/studentReporting.js [hunk 3]
```diff
@@ -36,37 +81,40 @@ Template.studentReporting.rendered = async function() {
...
-      drawCharts(studentData);
-    }
+    updateDashboard(selectedTdfId);
   },
 });
 
+async function updateDashboard(selectedTdfId){
+  console.log('change tdf select', selectedTdfId);
+  if (selectedTdfId!==INVALID) {
+    $(`#tdf-select option[value="${INVALID}"]`).prop('disabled', true);
```
Comment (andrewtackett) [inline mofacts/client/views/experimentReporting/studentReporting.js:103]: Are those smart/back ticks?  Can this be done with single quotes instead?

File: mofacts/server/methods.js [hunk 4]
```diff
@@ -1020,6 +1020,15 @@ async function getStimCountByStimuliSetId(stimuliSetId) {
   return ret.count;
 }
 
+async function getStimCountByTdfId(tdfId) {
```
Comment (andrewtackett) [inline mofacts/server/methods.js:1023]: This duplicates an existing function :P

**Review by andrewtackett — CHANGES_REQUESTED** [review]
A few minor changes.  Looks pretty good overall

**Comment by JRustyHaner** [comment]
@andrewtackett

**Review by andrewtackett — APPROVED** [review]

**Pull request merged**